src/Process/PostDeploy/WarmUp.php

```diff
@@ -44,63 +38,53 @@ class WarmUp implements ProcessInterface
      */
     private $logger;
 
-    /**
-     * Variable for caching base urls hosts from config table
-     *
-     * @var string
-     */
-    private $baseHosts;
-
     /**
      * @param PostDeployInterface $postDeploy
-     * @param ClientFactory $clientFactory
-     * @param RequestFactory $requestFactory
+     * @param PoolFactory $poolFactory
      * @param UrlManager $urlManager
      * @param LoggerInterface $logger
      */
     public function __construct(
         PostDeployInterface $postDeploy,
-        ClientFactory $clientFactory,
-        RequestFactory $requestFactory,
+        PoolFactory $poolFactory,
         UrlManager $urlManager,
         LoggerInterface $logger
     ) {
         $this->postDeploy = $postDeploy;
-        $this->clientFactory = $clientFactory;
-        $this->requestFactory = $requestFactory;
+        $this->poolFactory = $poolFactory;
         $this->urlManager = $urlManager;
         $this->logger = $logger;
     }
 
     /**
      * {@inheritdoc}
-     *
-     * @codeCoverageIgnore
+     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
      */
     public function execute()
     {
-        $client = $this->clientFactory->create();
-        $promises = [];
+        $urls = $this->getUrlsForWarmUp();
 
-        try {
-            foreach ($this->getUrlsForWarmUp() as $url) {
-                $request = $this->requestFactory->create('GET', $url);
-
-                $promises[] = $client->sendAsync($request)->then(function () use ($url) {
-                    $this->logger->info('Warmed up page: ' . $url);
-                }, function (RequestException $exception) use ($url) {
-                    $context = $exception->getResponse()
-                        ? [
-                            'error' => $exception->getResponse()->getReasonPhrase(),
-                            'code' => $exception->getResponse()->getStatusCode(),
-                        ]
-                        : [];
-
-                    $this->logger->error('Warming up failed: ' . $url, $context);
-                });
+        $fulfilled = function ($response, $index) use ($urls) {
+            $this->logger->info('Warmed up page: ' . $urls[$index]);
+        };
+
+        $rejected = function (RequestException $exception, $index) use ($urls) {
+            $context = [];
+
+            if ($exception->getResponse()) {
+                $context = [
+                    'error' => $exception->getResponse()->getReasonPhrase(),
+                    'code' => $exception->getResponse()->getStatusCode(),
+                ];
             }
 
-            \GuzzleHttp\Promise\unwrap($promises);
+            $this->logger->error('Warming up failed: ' . $urls[$index], $context);
+        };
+
+        try {
+            $pool = $this->poolFactory->create($urls, compact('fulfilled', 'rejected'));
```

Comment:
`compact` should be avoided

Reply:
Why? It's a safe and reliable method.

Reply:
It has broad visibility scope and makes code less readable

Reply:
I'm not sure what you mean by "visibility scope"; `compact()` doesn't have any greater scope than other variables/functions. I also fundamentally disagree that
```php
['fulfilled' => $fulfilled, 'rejected' => $rejected]
```
is somehow more readable.

Reply:
Answered on the previous `compact` usage